node.fetch ESM import issue in getFileByURL.js

### Link to reproduction

private repo

### Describe the Bug

Getting the following error in Payload 2.4.0 while trying to build the types:

```
/.../node_modules/payload/dist/uploads/getFileByURL.js:121
undefined
      ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /.../node_modules/node-fetch/src/index.js from /.../node_modules/payload/dist/uploads/getFileByURL.js not supported.
Instead change the require of index.js in /.../node_modules/payload/dist/uploads/getFileByURL.js to a dynamic import() which is available in all CommonJS modules.
    at Object.newLoader [as .js] (/.../node_modules/pirates/lib/index.js:121:7)
    at Object.<anonymous> (/.../node_modules/payload/dist/uploads/getFileByURL.js:11:59)
    at Object.newLoader [as .js] (/.../node_modules/pirates/lib/index.js:121:7) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v20.10.0

npm ERR! Lifecycle script `payload:types` failed with error: 
```

### To Reproduce

npm run payload:types is failing after upgrading to 2.4.0. All packages are the latest versions.

### Payload Version

2.4.0

### Adapters and Plugins

db-mongodb, bundler-webpack, live-preview

--- a/packages/payload/src/admin/hooks/useThumbnail.ts
+++ b/packages/payload/src/admin/hooks/useThumbnail.ts
@@ -48,6 +48,10 @@ const useThumbnail = (
       return `${pathURL}/${sizes[adminThumbnail].filename}`
     }
 
+    if (url) {
+      return url as string
+    }
+
     return `${pathURL}/${filename}`
   }
 

--- a/packages/payload/src/uploads/generateFileData.ts
+++ b/packages/payload/src/uploads/generateFileData.ts
@@ -67,7 +67,7 @@ export const generateFileData = async <T>({
     const { filename, url } = data as FileData
 
     try {
-      if (url && url.startsWith('/')) {
+      if (url && url.startsWith('/') && !disableLocalStorage) {
         const filePath = `${staticPath}/${filename}`
         const response = await getFileByPath(filePath)
         file = response as UploadedFile

--- a/packages/payload/src/uploads/getFileByURL.ts
+++ b/packages/payload/src/uploads/getFileByURL.ts
@@ -3,9 +3,9 @@ import path from 'path'
 import type { File } from './types'
 
 const getFileByURL = async (url: string): Promise<File> => {
-  const { default: fetch } = (await import('node-fetch')) as any
-
   if (typeof url === 'string') {
+    const { default: fetch } = (await import('node-fetch')) as any
+
     const res = await fetch(url, {
       headers: {
         'Content-Type': 'application/json',

--- a/packages/payload/src/uploads/getFileByURL.ts
+++ b/packages/payload/src/uploads/getFileByURL.ts
@@ -2,13 +2,8 @@ import path from 'path'
 
 import type { File } from './types'
 
-const importDynamic = (modulePath: string) => import(modulePath)
-
 const getFileByURL = async (url: string): Promise<File> => {
-  async function fetch(...args) {
-    const { default: fetch } = await importDynamic('node-fetch')
-    return fetch(...args)
-  }
+  const { default: fetch } = (await import('node-fetch')) as any
 
   if (typeof url === 'string') {
     const res = await fetch(url, {

--- a/packages/payload/src/uploads/getFileByURL.ts
+++ b/packages/payload/src/uploads/getFileByURL.ts
@@ -1,24 +1,21 @@
-import fetch from 'node-fetch'
 import path from 'path'
 
 import type { File } from './types'
 
 const getFileByURL = async (url: string): Promise<File> => {
   if (typeof url === 'string') {
-    const res = await fetch(url, {
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      method: 'GET',
-    })
-    const data = await res.buffer()
+    const res = await fetch(url)
+    const blob = await res.blob()
+    const arrayBuffer = await blob.arrayBuffer()
+
+    const data = Buffer.from(arrayBuffer)
     const name = path.basename(url)
 
     return {
       name,
       data,
-      mimetype: res.headers.get('content-type') || undefined,
-      size: Number(res.headers.get('content-length')) || 0,
+      mimetype: blob.type || undefined,
+      size: blob.size || 0,
     }
   }
 }

--- a/packages/payload/src/uploads/getFileByURL.ts
+++ b/packages/payload/src/uploads/getFileByURL.ts
@@ -2,20 +2,29 @@ import path from 'path'
 
 import type { File } from './types'
 
-const getFileByURL = async (url: string): Promise<File> => {
-  if (typeof url === 'string') {
-    const res = await fetch(url)
-    const blob = await res.blob()
-    const arrayBuffer = await blob.arrayBuffer()
+const importDynamic = (modulePath: string) => import(modulePath)
 
-    const data = Buffer.from(arrayBuffer)
+const getFileByURL = async (url: string): Promise<File> => {
+  async function fetch(...args) {
+    const { default: fetch } = await importDynamic('node-fetch')
+    return fetch(...args)
+  }
+
+  if (typeof url === 'string') {
+    const res = await fetch(url, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'GET',
+    })
+    const data = await res.buffer()
     const name = path.basename(url)
 
     return {
       name,
       data,
-      mimetype: blob.type || undefined,
-      size: blob.size || 0,
+      mimetype: res.headers.get('content-type') || undefined,
+      size: Number(res.headers.get('content-length')) || 0,
     }
   }
 }

--- a/packages/payload/src/uploads/imageResizer.ts
+++ b/packages/payload/src/uploads/imageResizer.ts
@@ -234,7 +234,7 @@ export default async function resizeAndTransformImageSizes({
       }
 
       const imageToResize = sharpBase.clone()
-      let resized = imageToResize.resize(imageResizeConfig)
+      let resized = imageToResize
 
       if (
         req.query?.uploadEdits?.focalPoint &&
@@ -283,6 +283,8 @@ export default async function resizeAndTransformImageSizes({
           top: safeOffsetY,
           width: safeResizeWidth,
         })
+      } else {
+        resized = imageToResize.resize(imageResizeConfig)
       }
 
       if (imageResizeConfig.formatOptions) {

